airflow/api_fastapi/core_api/datamodels/dag_run.py

```diff
@@ -72,3 +72,13 @@ class DAGRunCollectionResponse(BaseModel):
 
     dag_runs: list[DAGRunResponse]
     total_entries: int
+
+
+class TriggerDAGRunPostBody(BaseModel):
+    """Trigger DAG Run Serializer for POST body."""
+
+    dag_run_id: str
+    data_interval_start: datetime | None
+    data_interval_end: datetime | None
```

Comment:
According to [docs](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/post_dag_run) both start and end data interval are nullable. I noticed in [legacy impl](https://github.com/apache/airflow/blob/1a3a96cac7ab3996bc1580ec3dfef20f5c753a66/airflow/api_connexion/schemas/dag_run_schema.py#L115), both of these are required. Else, we raise 400 status. If they are required, can we make them non-nullable?

Reply:
No, it means that they both must be defined or not defined. You can't just send one but not the other.